doc/ch-fromhost_desc.rst

```diff
@@ -116,57 +145,73 @@ some use cases and the recommended approach:
-      Use the :code:`shrink-soname` branch. (MPICH only.)
-
-   c. :code:`libgfortran.so.3`, because Cray's :code:`libmpi.so.12`
-      links to it. (MPICH only.)
-
-2. Applications must be dynamically linked to :code:`libmpi.so.12` (not e.g.
-   :code:`libmpich.so.12`).
-
-   a. How to configure MPICH to accomplish this is not yet clear to us;
-      :code:`test/Dockerfile.mpich` does it, while the Debian packages do not.
-      (MPICH only.)
-
-3. An ABI compatible module for the given MPI implementation must be loaded
-   when :code:`ch-fromhost` is invoked.
-
-   a. Load the :code:`cray-mpich-abi` module. (MPICH only.)
-
-   b. We recommend loading the module of a version as close to what
-      is installed in the image as possible. This OpenMPI install needs to be
-      built such that libmpi contains all needed plugins (as opposed to them
-      being standalone shared libraries). See `OpenMPI's documentation
-      <https://www.open-mpi.org/faq/?category=building>`_ for how to do this.
-      (OpenMPI only.)
+   a. Libfabric (https://ofiwg.github.io/libfabric/). See
+      :code:`charliecloud/examples/Dockerfile.libfabric`.
+
+   b. Corresponding open source MPI implementation configured and built against
+      the container libfabric, e.g.,
+      - `MPICH <https://www.mpich.org/>`_, or
+      - `OpenMPI <https://www.open-mpi.org/>`_.
+      See :code:`charliecloud/examples/Dockerfile.mpich` and
+      :code:`charliecloud/examples/Dockerfile.openmpi`.
+
+2. The Cray UGNI loadable provider, :code:`libgnix-fi.so`, will link to
+   compiler(s) in the programming environment by default. For example, if it
+   is built under the :code:`PrgEnv-intel` PE, the provider will have links to
+   files at paths :code:`/opt/gcc` and :code:`/opt/intel` that :code:`ch-run`
+   will not bind automatically.
+
+   Managing all possible bind-mount paths is untenable. Thus, this experimental
+   implementation works only with Cray UGNI provider(s) built on XC series
+   systems with the minimal modules necessary to compile provider and
+   leverage the Aries interconnect at run-time, e.g.,:
+
+   - modules
+   - craype-network-aries
+   - eproxy
+   - slurm
+   - cray-mpich
+   - craype-haswell
+   - craype-hugepages2M
+
+   Cray UGNI providers linked against more complicated PE's will work assuming
+   1) the user explicitly bind-mounts any and all missing paths from the
+   provider's :code:`ldd` output, and 2) all such paths do not conflict with
+   container functionality, e.g., :code:`/usr/bin/`, etc.
 
 4. Tested only for C programs compiled with GCC, and it probably won't work
-   otherwise. If you'd like to use another compiler or another programming
-   language, please get in touch so we can implement the necessary support.
+   without extensive bind-mounts and kluding. If you'd like to use another
+   compiler or programming environment, please get in touch so we can implement
+   the necessary support.
```

Comment:
We should support the compilers our code teams care about, at least. I think Intel is big?

Reply:
AFIACT, container access to the Aries interconnect via the host's injected `ugni` provider is not dependent on the container's compiler(s). For example, if the container was built with intel MPI against libfabric, it should still be able to pick up and use this same host `libgnix-fi.so` compiled without references to `/opt/intel`.

Reply:
OK. You might include that explanation in the docs as well.